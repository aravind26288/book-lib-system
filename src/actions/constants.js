export const bookDetails = {
    ADD_BOOK: 'ADD_BOOK',
    SEARCH_BOOK: 'SEARCH_BOOK'
}