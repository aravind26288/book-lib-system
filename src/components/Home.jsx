import React from 'react';
import {
    Container,
    Row,
    Col,
    Form,
    FormGroup,
    Input} from 'reactstrap';
import {connect} from 'react-redux';

export class Home extends React.PureComponent{
    constructor(props){
        super(props);
        this.state = {
            visibleBooks: [],
            searchFlag: false 
        }
        this.searchBooks = this.searchBooks.bind(this);
    }
    searchBooks(e){
        let searchTerm = e.target.value;
        if(searchTerm.trim() !== ''){
            let filteredBooks = this.props.books.filter(function(book, index){
                return (book.bookName.toLowerCase().includes(searchTerm.toLowerCase())  || book.bookDesc.toLowerCase().includes(searchTerm.toLowerCase()) || book.bookAuthor.toLowerCase().includes(searchTerm.toLowerCase()));
            });

            this.setState({
                visibleBooks: filteredBooks,
                searchFlag: true
            });
        }
        else{
            let filteredBooks = this.props.books.filter(function(book, index){
                return '';
            });

            this.setState({
                visibleBooks: filteredBooks,
                searchFlag: false
            });
        }
    }
    render(){
        return (
            <React.Fragment>
            <Container>
                <Row className="mt-3">
                    <Col className="book-lib-app__content-wrapper">
                    <Row className="mt-2">
                        <Col><div className="book-lib-app__content-title">Home</div></Col>
                        </Row>
                    <Row className="mt-5">
                        <Col lg={{size: 6, offset: 3}} md={{size: 8, offset: 2}} sm="12" xs="12">
                        <Form>
                          <FormGroup>
                              <Input type="text" className="searchBook" name="search-book" id="search-book" placeholder="Search book by name, author..." onKeyUp={this.searchBooks}/>
                          </FormGroup>
                      </Form>
                        </Col>
                    </Row>
                    <Row className="mt-3">
                        <Col lg={{size: 6, offset: 3}} md={{size: 8, offset: 2}} sm="12" xs="12">
                        {Array.isArray(this.state.visibleBooks) && this.state.searchFlag === true && this.state.visibleBooks.length === 0 &&
                         <Row className="book-lib-app__book">
                             <Col>
                             <div>
                         No results found
                     </div>
                             </Col>
                         </Row>
                         }
                         {this.state.visibleBooks.map((books) => 
                         <Row key={books.bookId} className='book-lib-app__book'>
                             <Col>
                                <div><span className="book-lib-app__book-txtLabel">Book name:</span> {books.bookName}</div>
                                <div><span className="book-lib-app__book-txtLabel">Description:</span> {books.bookDesc}</div>
                                <div><span className="book-lib-app__book-txtLabel">Author:</span> {books.bookAuthor}</div>
                             </Col>
                         </Row>
                         )}
                        </Col>
                    </Row>
                    </Col>
                </Row>
            </Container>
            </React.Fragment>
        )
    }
}

const mapStateToProps = state => ({
    books: state.books
});

export default connect(mapStateToProps, null)(Home);