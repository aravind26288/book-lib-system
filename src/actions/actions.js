import {bookDetails} from './constants';

let bookId = 1;
export const addBookDetails = (book) => {
    return {
    type: bookDetails.ADD_BOOK,
    id: bookId++,
    book
    }
};

export const searchBook = (searchTerm) => ({
    type: bookDetails.SEARCH_BOOK,
    searchTerm
})