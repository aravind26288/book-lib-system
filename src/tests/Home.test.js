import React from 'React';
import Home from '../components/Home';
import Enzyme, {mount} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {addBookDetails} from '../actions/actions';
import {Provider} from 'react-redux';
import {bookDetails} from '../actions/constants';
import Adapter from 'enzyme-adapter-react-16';
const mockStore = configureMockStore();
Enzyme.configure({adapter: new Adapter()});
const initialAppState = {
    books:[]
};
describe('home page', () => {
    it('home page renders a search box', () => {
        let state;
        const store = mockStore(initialAppState);
        const homeComp = mount(<Provider store={store}><Home /></Provider>);
        store.dispatch(addBookDetails({bookname: 'B 1', bookdesc: 'B 1 DESC', bookauthor: 'Aravind'}));
        store.dispatch(addBookDetails({bookname: 'B 2', bookdesc: 'B 2 DESC', bookauthor: 'Accenture'}));
        store.dispatch(addBookDetails({bookname: 'B 3', bookdesc: 'B 3 DESC', bookauthor: 'Aravind B'}));
        let searchInput = 'Accenture';
        const searchInputComp = homeComp.find('#search-book').first();
        searchInputComp.simulate('keyup', {target: {name: 'search-book',value: searchInput}});
        state = store.getState();
         expect(homeComp).toMatchSnapshot();
        })
})