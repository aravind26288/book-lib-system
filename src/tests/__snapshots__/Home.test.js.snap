// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page home page renders a search box 1`] = `ReactWrapper {}`;
